### PR TITLE
Split conflicting overloads as WebGLXRenderingContextOverloads

### LINKED ISSUE #2216: IDL for WebGL2RenderingContext is not valid: has multiple identical overloads of bufferData
The relevant bits are, from https://www.khronos.org/registry/webgl/specs/latest/2.0/webgl2.idl and https://www.khronos.org/registry/webgl/specs/latest/1.0/webgl.idl :

    [NoInterfaceObject]
    interface WebGL2RenderingContextBase
    {
    ...
      // WebGL1:
      void bufferData(GLenum target, GLsizeiptr size, GLenum usage);
      void bufferData(GLenum target, ArrayBuffer? srcData, GLenum usage);
      void bufferData(GLenum target, ArrayBufferView srcData, GLenum usage);
    ...
      // WebGL2:
      void bufferData(GLenum target, ArrayBufferView srcData, GLenum usage, GLuint srcOffset,
                      optional GLuint length = 0);
    ...
    };

    WebGL2RenderingContextBase implements WebGLRenderingContextBase;

    [NoInterfaceObject]
    interface WebGLRenderingContextBase
    {
      void bufferData(GLenum target, GLsizeiptr size, GLenum usage);
      void bufferData(GLenum target, ArrayBuffer? data, GLenum usage);
      void bufferData(GLenum target, ArrayBufferView data, GLenum usage);
    };

The "webgl1" versions of bufferData are present twice: once directly and once pulled in via the `implements` statement.  That leads to there being identical overloads, which is not valid IDL.

There are similar problems, at first glance, with bufferSubData, texImage2D, etc.

### PR DESCRIPTION
**Pull request by saschanaz**
Fixes #2216

### CODE REVIEW

File: sdk/tests/py/lint/lint.py [hunk 1]
```diff
@@ -157,19 +156,19 @@ def check_regexp_line(path, f):
 
 def output_errors(errors):
     for error_type, error, line_number in errors:
-        print "%s: %s" % (error_type, error)
+        print("%s: %s" % (error_type, error))
```
Comment (jdashg) [inline sdk/tests/py/lint/lint.py]: Are you using Arch Linux? These look like python3 compatibility changes.

Please split these into a different PR, and modify the shebang line to either python2 or python3.

Comment (saschanaz) [reply sdk/tests/py/lint/lint.py]: #2865

**Comment by jdashg** [comment]
Thanks!
